### PR TITLE
CI: add github action to post to first-time PRs openers

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

Not sure how to test this?  

Below the line is how I think this should render:

--------

Thank you for opening your first PR into Matplotlib!

If you have not heard from us in a while please feel free to ping `@matplotlib/developers` or anyone who has commented on the PR.  Most of our reviewers are volunteers and it is easy for things to fall through the cracks.

You can also join us [on gitter](https://gitter.im/matplotlib/matplotlib) if real-time discussion would be useful.

For details on testing, write docs, and our review process please see [the developer guide](https://matplotlib.org/devdocs/devel/index.html)

We strive to be a welcoming and open project, please follow our [Code of Conduct](https://github.com/matplotlib/matplotlib/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: .github/workflows/pr_welcom.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+          If you have not heard from us in a while please feel free to
+          ping `@matplotlib/developers` or anyone who has commented on the
+          PR.  Most of our reviewers are volunteers and it is easy for
+          things to fall through the cracks.
+
+          You can also join us [on
+          gitter](https://gitter.im/matplotlib/matplotlib) if real-time
+          discussion would be useful.
+
+          For details on testing, write docs, and our review process
```
Comment (story645) [inline .github/workflows/pr_welcom.yml:25]:
```suggestion
          For details on testing, writing docs, and our review process
```

**Review by story645 — CHANGES_REQUESTED** [review]
Typo. But also where is the list this generates of PRs this would be applied to?

File: .github/workflows/pr_welcom.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: PR Greetings
```
Comment (QuLogic) [inline .github/workflows/pr_welcom.yml:1]: Typo in file name: `pr_welcom.yml`

**Comment by tacaswell** [comment]
I do not think there is a list anyplace, this should run on any PR opened by a first-time contributor and leave the welcome message.

**Comment by ianhi** [comment]
> Not sure how to test this?

Maybe merge this into your fork's default branch and then have someone open a PR against that?

Also this is a nice thing :+1: to is, especially to the explicit language making clear that it's ok to ping people if it lies dormant.

**Comment by ianhi** [comment]
Maybe also include a mention of https://gitter.im/matplotlib/incubator? This seems like part of the audience you're trying to capture?

File: .github/workflows/pr_welcome.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+    - uses: actions/first-interaction@v1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        pr-message: >+
+          Thank you for opening your first PR into Matplotlib!
+
+          If you have not heard from us in a while please feel free to
+          ping `@matplotlib/developers` or anyone who has commented on the
+          PR.  Most of our reviewers are volunteers and it is easy for
```
Comment (timhoffm) [inline .github/workflows/pr_welcome.yml:18]:
```suggestion
          PR.  Most of our reviewers are volunteers and sometimes
```

Sounds a little more professional to me 😃 .

File: .github/workflows/pr_welcome.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+          things to fall through the cracks.
+
+          You can also join us [on
+          gitter](https://gitter.im/matplotlib/matplotlib) if real-time
+          discussion would be useful.
```
Comment (timhoffm) [inline .github/workflows/pr_welcome.yml:23]:
```suggestion
          gitter](https://gitter.im/matplotlib/matplotlib) for real-time
          discussion.
```

**Review by story645 — DISMISSED** [review]

**Review by jklymak — APPROVED** [review]

**Review by timhoffm — APPROVED** [review]
Concerning testing, I'd just wait for the next new contributor and see if the action triggers properly. :smile:

**Comment by timhoffm** [comment]
Ping @tacaswell : This does not work.

See e.g. #19172.

https://github.com/matplotlib/matplotlib/pull/19172/checks?check_run_id=1606582798

**Comment by tacaswell** [comment]
I think I was too optomistic about the secrets already being set up.  I knew @QuLogic had set one up from the wheels but it looks like it is the wrong name for his to discover and I suspect has the permissions locked down enough that it would not work anyway...

**Comment by tacaswell** [comment]
xref https://github.com/actions/first-interaction/issues/10

I'm now confused if we can actually make this work?  If it needs access to a secret to post, then that means the secret needs to be available to the PR, which means it is extract able via a malicious PR that adds a task to print out said secret, which means we can not give the action access to the secret, which means the action can not post back to the thread :facepalm: .

**Comment by jklymak** [comment]
I thought this was an email to the user, not posted to the PR.

**Comment by ufmayer** [comment]
On my first PR to the matplotlib project I did not receive a welcome email, instead my CI build failed. Digging through the logs I found a reference to the welcome message for first-time contributors as a FAIL during the build. It's posted on that PR #19172 that @timhoffm already linked above in this discussion.

**Comment by jklymak** [comment]
@ufmayer Ooops, sorry we made you the guinea pig! This was something new we were trying.  Hopefully it is working now.

**Comment by ufmayer** [comment]
No problem. If I can help out here let me know, but I guess I can make my first PR only once, and that's done for.

**Comment by jklymak** [comment]
The welcome message is still not working?  

https://github.com/matplotlib/matplotlib/pull/19187/checks?check_run_id=1619669692

**Comment by dopplershift** [comment]
Tried to fix it, see #19196.

**Pull request merged**